ement-room.el

```diff
@@ -2817,151 +2809,113 @@ MSG is the message STRING was contained in."
-                   'url-preview-n)
-        ;; (< num (with-temp-buffer
-        ;;          (insert string)
-        ;;          (goto-char (point-min))
-        ;;          (cl-loop with in-preview? = nil
-        ;;                   while (< (point) (point-max))
-        ;;                   do (if (looking-at-p (concat "^" (regexp-quote ement-room-url-preview-prefix) ".*"))
-        ;;                          (setf in-preview? t)
-        ;;                        (setf in-preview? nil))
-        ;;                   if in-preview?
-        ;;                   count (looking-at-p "^\n")
-        ;;                   do (beginning-of-line 2))))
-        ;; )
-        (ement-api ement-session "preview_url"
-          :endpoint-category "media" :params `((url ,link))
-          :then (lambda (data)
-                  (ement-room--insert-preview buf msg related-event data num))
-          :else (lambda (why)
-                  (ement-debug "Error in fetching preview: " why)))))))
-
-;; (defun ement-room--insert-preview (buf msg data num)
-;;   "Insert link preview for DATA into MSG, if it isn't already present.
-;; BUF is the room buffer this link is from. NUM is the index for this link."
-;;   (cond ((alist-get 'errcode data) (message "Ement: You are being rate limited."))
-;;         (data (with-current-buffer buf
-;;                 (let* ((node (ement-room--ewoc-node-before ement-ewoc
-;;                                                            msg
-;;                                                            (lambda (a b)
-;;                                                              (string= (ement-event-id a)
-;;                                                                       (ement-event-id b)))
-;;                                                            :pred #'ement-event-p))
-;;                        (next? (cl-loop with w = (ewoc-next ement-ewoc node)
-;;                                        repeat num
-;;                                        if (or (not w)
-;;                                               (not (listp (ewoc-data w)))
-;;                                               (not (eq (car (ewoc-data w))
-;;                                                        'url-preview))
-;;                                               (not (equal (cadr (ewoc-data w))
-;;                                                           data)))
-;;                                        return t
-;;                                        do (setf w (ewoc-next ement-ewoc w))
-;;                                        finally return nil)))
-;;                   (when next?
-;;                     (with-silent-modifications
-;;                       (ewoc-enter-after ement-ewoc
-;;                                         node
-;;                                         `(url-preview ,data)))))))))
-
-(defun ement-room--insert-preview (buf msg related data num)
+(defun ement-room--get-preview (msg link num)
```

Comment:
Allow me to get a bit nit-picky: names are important.  :)  `get` is an anti-pattern, IMHO.  Instead, just name the function for what it returns or does.  Please see the way other functions are named in this file.

As well, please follow the conventions around argument names (i.e. they should be clear and usually unabbreviated) and docstrings (e.g. "Return foo for BAR.").

Reply:
Sure thing :+1: